### PR TITLE
[TIMOB-23529] Build: Replace scons and python with node scripts

### PR DESCRIPTION
**Pull request by sgtcoolguy**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23529

I'd like to replace our build scripts in-place, so I'd prefer to get the build working on our PR build and then hopefully merging will just magically replace our build system.
To do so, I still need to take our existing build steps that run for titanium_mobile_master and migrate them into a bash script that lives in the repo itself - so that this PR can then overwrite part of that script to run our new node scripts instead of scons.

This PR does not yet replace the downloading of windows for builds, or uploading the built zips for all OSes up to S3. It also assumes you have Xcode/Android SDK/Android NDK, etc installed (i.e. I didn't write a script to fetch and set up pre-requisites yet).

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
@cb1kenobi Hey Chris, looks like this PR is finally set to be reviewed. Thanks!

File: build/packager.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	this.srcDir = ROOT_DIR;
+	this.outputDir = outputDir; // root folder where output is placed
+	this.targetOS = targetOS;
+	this.platforms = platforms;
+	this.version = version;
+	this.versionTag = versionTag;
+	this.moduleApiVersion = moduleApiVersion;
+	this.gitHash = gitHash;
+	var date = new Date();
+	this.timestamp = '' + (date.getMonth() + 1) + '/' + date.getDate() + '/' + (date.getYear()) + ' ' + date.getHours() + ':' + (date.getMinutes());
```
Comment (cb1kenobi) [inline build/packager.js]: Use `date.getFullYear()`.

Comment (sgtcoolguy) [inline build/packager.js]: All right, just pushed that plus the left pad of hours and minutes with '0'.

**Comment by cb1kenobi** [comment]
**APPROVED**

**Pull request merged**